[GM Bounty] Space Station 13. Home. At least it was before I fucked everything up.

Iamgoofball opened the pull request:
@vuonojenmustaturska @Robustin
:cl: Iamgoofball, Robustin
add: A new gamemode is hitting the station, Families.
add: In this gamemode, 3 different crime families will be spawned on board the station. At the 2 minute mark, it will be announced, and a 30 minute timer will start.
add: During these 30 minutes, all 3 crime families need to work to gain as many Points as possible.
add: Joining a gang is 100% voluntary. There's no forced conversion. To join a family, simply open hand palm slap the Signup Point.
add: Points can be gained by tagging turf, holding turf, wearing your colors, convincing civilians or rival gangsters to join up or just wear the colors, selling contraband on the black market, rolling around in a crew, and other various methods of criminal activity!
add: At the end of the 30 minutes, the game ends, and the gang with the most points is declared the winner!
experimental: This gamemode will require 30 players.
/:cl:

Feedback thread:
https://tgstation13.org/phpBB/viewtopic.php?f=10&t=19518

Supplementary Video:
https://www.youtube.com/watch?v=OuvUgaLxemg

**Isn't this just Gang?**
Fuck no. Only thing similar is tagging turf and the fact gangs are involved. This mode is completely different otherwise.

**Shit, where do I sign up?**
Ask a family member where their Signup Point is, and then simply click on it with an open hand. You'll be signed up for the family instantly, and given some sick threads and a spraycan for tagging.

**How do I rep my family?**
Wear your gang's uniform or colors similar.

**What does it mean to roll with a crew?**
Travel in a group of four or more for bonus points towards your gang.
However, you receive less for having eight or more, so be careful. Try to spread your crews out!

**I'm an X, what do?**
Gangster: Yeah, go do whatever. Wanna backstab your gang? Go for it. You can switch sides at any time by clicking on an enemy gang's sign up point. Wanna murder some snitch because they ratted you out to the pigs? Do it. Wanna pressure the locals into supplying you with goods to export? Emergent gameplay.
Security: They can join up with gangs. Nothing is stopping them, implants aren't involved at all. This gamemode is intended to see how well players organize and stay united, or if they backstab eachother. It's all emergent gameplay.
Civillians: Wanna join a gang? Go for it. Gangsters probably shouldn't be arbitrarily murdering you, but if you're repping someone else's colors, don't expect to be given a free pass. After all, uniforms are the only way to really identify a gang member.

**No huds? How can I tell if someone's part of my group?**
Tough shit, man. I hope you like trust.

In short, this gamemode will be a fun exercise in how far the playerbase can go in regards to trusting eachother and unifying to meet a common goal for their group with no rules, gameplay mechanics, or anything actively forcing them to work together.

**I'm an admin, what do I need to know about the valids?**
Gangster VS Gangster: Valid. All times. Even if it's same-team. These crime families aren't exactly known for their stability or rigid code of honor.
Gangster VS Civvie: Don't start shit for no good reason. Standard escalation rules apply.
Civvie VS Gangster: Same as above.
Security: Security IS civvies for all intents and purposes. Unless they join up with a gang.

**How do I know if someone is a gangster?**
1. Are they wearing a gang uniform/color?
If yes, they're probably a gangster.
If no, they're probably a civvie.

Iamgoofball committed:
Space Station 13. Home. At least it was before I fucked everything up.

Iamgoofball committed:
its not gangs i swear

Iamgoofball committed:
whoops

Iamgoofball committed:
double whoops

optimumtact commented:
aww gee whiz bill, why does your codebase let you have TWO GANGS?

Iamgoofball commented:
i guess its a good thing we only have 1 gang, and it's this PR

Mickyan commented:
have we learned nothing from war ops about announcing the gamemode at roundstart

Iamgoofball commented:
@Mickyan 
War Ops is a gamemode about 5 people showing up to murder everyone and nuke the station.

This is not a gamemode about 5 people showing up to murder everyone and nuke the station.

This is a gamemode about teamwork.

Iamgoofball commented:
>build failed

well lets go check what I forgot to com-

![](https://file.house/9BOC.png)

# REMOVE TRAVIS

Iamgoofball committed:
More colors

Carbonhell commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -0,0 +1,164 @@
+/datum/game_mode/gang
```
**it's not gang**

Iamgoofball replied:
i'll change this on the codeside soon

Borisvanmemes commented:
This reminds me of the gang gamemode that goon has, might be fun!

Bawhoppen commented:
Gang 2.0: It's gonna work this time edition

Iamgoofball committed:
Families Balance Pass 1

Grazyn commented:
@Bawhoppen but that wasn't _real_ gang

also what's stopping all sec from going gang and using their powerful gear to steamroll the entire station

Iamgoofball commented:
@Grazyn Since the round does not end until 60 minutes has passed, there is no reason to blitzkrieg or rush to win.

Grazyn commented:
Eh, it's not like players always do things "for a reason". Some have fun from making other people miserable. And rules says that antag can do whatever they want.

Also, can't you call the shuttle to end the round early? Or does it only end after 60 minutes

Therapist1 commented:
I like it, but remember that players may take the concept in the wrong direction, such as Grazyn’s security example, but I do like the idea of medical becoming a gang’s citadel/drug lab.

Iamgoofball committed:
Way, way shorter roundtimes. No murderbone/mass destruction.

Iamgoofball commented:
updated rules as of 9:27 AM
OOC ":singulo: alright, updated meme rules for the next run of Families in a few hours probably idk whenever I wake the fuck up https://tgstation13.org/phpBB/viewtopic.php?f=10&t=19518#p438533 :singulo:

davethwave commented:
Well, hope this turns out well since it looks interesting.

PKPenguin321 commented:
Changelog says the timer is both 30 and 60 minutes. Which is it?

Iamgoofball commented:
3o minutes

ExcessiveUseOfCobblestone commented:
This is extended with flavor, probably should replace extended and the timer dropped tbh :^)

HydroFloric commented:
Gotta say goof i wasnt expecting something like this from you. Maybe im just biased from your reputation but this looks interesting reminds me of goons gang mode. Of course we need to see it in practice first. Also side not how is family naming going to work?

ExcessiveUseOfCobblestone commented:
If it's not input then I hate this mode

GuppyLaxx commented:
this is just goon gangs lmao

obscolene commented:
<img src="https://i.gyazo.com/19b586699a869ba181e293a223d68782.png">

i'm havin a big laffo @ ur expense rn chief

Cyberboss commented:
If this isn't a meme, we should have mandatory balancing of the factions

Iamgoofball commented:
Could you explain? Or do you mean numbers wise?

Cyberboss commented:
Like a no team should be able to get more than x number of players than the others (probably should be a config option). I.e. not everyone can join team A, leaving teams B and C with 1 member

ExcessiveUseOfCobblestone commented:
If you see one gang dominating you aren't going to want to be curbstomped because you have to join one of the other two though.

Supermichael777 commented:
This seems more like a side antag thing,  in a long round have a flare up in gang tensions and pull some dopes back into the life. also needs more to prevent MSS13 every time,  ought-right murder sprees need to be actively discouraged.

Grazyn commented:
Automatic balancing like in battlefield? If teams are unbalanced, players are randomly switched between teams.

ExcessiveUseOfCobblestone commented:
Why does it have to be balanced? If everyone wants to win then everyone can win and it can be pseudo-extended.

The whole point of this mode from my end at least is that it's meant to be sandboxy extended with the mini-antag bit as an enabler. Players are suppose to formulate the path themselves.

epochayur commented:
so, it's gangs?

Iamgoofball committed:
Goofgangs 3.0: Balance and Locations

adamkad1 commented:
ban sec for duration of the round, they cant hold their validhunting urges

Iamgoofball committed:
Tweak

Iamgoofball committed:
whoops

factoryman942 commented:
If two gang signups are sufficiently close to each other and both gangs have space you can simply click each one and spawn a bunch of excess suits and hats and sprays
maybe add a short cooldown to it or something, like 5s

AnturK reviewed (changes requested):
Team construction needs changes.

Iamgoofball commented on code/modules/antagonists/gang/gang.dm:
```diff
@@ -0,0 +1,132 @@
+	name = "Family Member"
+	roundend_category = "gangsters"
+	var/gang_name = "Leet Like Jeff K"
+	var/gang_id = "LLJK"
```
I figured I'd end up using it again at some point for internal stuff, so I made the name generic enough to fit.

Iamgoofball commented on code/game/gamemodes/gang/gang_things.dm:
```diff
@@ -0,0 +1,67 @@
+	name = "Gang Signup Point"
+	icon = 'icons/obj/gang/signup_points.dmi'
+	max_integrity = INFINITY
+	obj_integrity = INFINITY
```
Where are they located at?

AnturK replied:
https://github.com/tgstation/tgstation/blob/12606763e9123a17261c4e62f30027c8a97ec1f4/code/__DEFINES/flags.dm#L63 and all the others in there..

Iamgoofball commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -0,0 +1,177 @@
+	var/highest_gang = "Leet Like Jeff K"
+	report += "<span class='header'>The gangs in the round were:</span>"
+	for(var/datum/team/gang/G in gangs)
+		report += "<span class='header'>[G.name]:</span>"
```
I scrapped team report in favor of this setup so that I could make sure the formatting was in the correct order for gangs. I also need to re-do the formatting anyways and order it from highest/lowest points

AnturK replied:
This locks this hard to the mode, should instead mark the participating teams as no show if you want to handle them yourself.

Iamgoofball replied:
A lot of the gang item stuff relies heavily on gang being the active gamemode, and this isn't intended to be a mulligan gamemode.

ExcessiveUseOfCobblestone replied:
It needs to be modularized enough for that to be a possibility though.

Iamgoofball replied:
Unless admin tools are able to call pre/post setup, that can't happen.

ExcessiveUseOfCobblestone replied:
that can't happen _in it's current form_ :^)

Iamgoofball commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -0,0 +1,177 @@
+		if (!antag_candidates.len)
+			break
+		var/datum/mind/gangbanger = antag_pick(antag_candidates)
+		gangbangers += gangbanger
```
I thought special_role got killed with antag datums

optimumtact commented:
goof you're basically playing dodgeball with all feedback on this mode so I don't see how you're possibly going to get it merged.

octareenroon91 commented:
>goof you're basically playing dodgeball with all feedback on this mode so I don't see how you're possibly going to get it merged.

Yeah, just refactor modes and roles into json configs then post this gangs on pastebin or something.

Iamgoofball commented:
@optimumtact I'm only playing dodgeball with people giving me bogus crap like "muh playerbase" and "validhunting". Things I can actually change mechanically, I will. This gamemode will be finished.

PKPenguin321 commented:
i was gonna say goof was full of shit but i actually agree with him, see https://tgstation13.org/phpBB/viewtopic.php?f=10&t=19518#p438933 (particularly his response to beesting). if he actually delivers on the features he mentions in that post then this could actually be shaping up to be a pretty fun mode. hope it can grow into something we can all enjoy goof, keep it up
